src/cert.rs

```diff
@@ -132,6 +132,10 @@ pub(crate) fn parse_cert_internal<'a>(
 // mozilla::pkix supports v1, v2, v3, and v4, including both the implicit
 // (correct) and explicit (incorrect) encoding of v1. We allow only v3.
 fn version3(input: &mut untrusted::Reader) -> Result<(), Error> {
+    if !input.peek(der::Tag::ContextSpecificConstructed0 as u8) {
```

Comment:
Please revert back to what you had before, where we just changed the error passed to `der::nested` below. We can adjust the description of `UnsupportedCertVersion` to indicate that it is returned if the version field missing, malformed, or an unsupported version.